Kotlin/lib/src/main/java/com/wolt/blurhashkt/BlurHashDecoder.kt

```diff
@@ -2,14 +2,25 @@ package com.wolt.blurhashkt
 
 import android.graphics.Bitmap
 import android.graphics.Color
-import kotlin.math.PI
 import kotlin.math.cos
 import kotlin.math.pow
 import kotlin.math.withSign
 
 object BlurHashDecoder {
 
-    fun decode(blurHash: String?, width: Int, height: Int, punch: Float = 1f): Bitmap? {
+    // cache Math.cos() calculations to improve performance.
+    // The number of calculations can be huge for many bitmaps: width * height * numCompX * numCompY * 2 * nBitmaps
+    // the cache is enabled by default, it is recommended to disable it only when just a few images are displayed
+    private val cacheCosinesX = HashMap<Int, DoubleArray>()
+    private val cacheCosinesY = HashMap<Int, DoubleArray>()
+
+    fun clearCache() {
+        cacheCosinesX.clear()
+        cacheCosinesY.clear()
+    }
```

Comment:
I see the cache is being clear only in test, so I have a few questions regarding this function:
- Can the test work without clearing the cache? Would this create some kind of cache conflict when decoding multiple image with the same `width * numCompX` and `height * numCompY`?
- If it does need clearing, when should user clear it?

Reply:
the idea is to use it when the app needs more memory. IMHO any controller using caches should include an option to clear it.

Reply:
This should be documented somewhere along side with the `useCache` param as well. Maybe a partial kdoc for `decode()`?

Reply:
it is a good idea, will do